### PR TITLE
Ts lint

### PR DESCRIPTION
**Pull request by jasiek-net**

### CODE REVIEW

File: src/helpers/passport.ts [hunk 1]
```diff
@@ -1,55 +1,51 @@
...
-        if (err || !user) {
-          reject(info)
-        }
-        resolve(user)
+    passport.authenticate(
+      'jwt',
+      { session: conf.jwt.session },
+      (err, user, info) => {
+        if (err || !user) return reject(info);
+        resolve(user);
```
Comment (michalmikolajczyk) [inline src/helpers/passport.ts]: Please return the promise
https://pouchdb.com/2015/05/18/we-have-a-problem-with-promises.html

File: src/helpers/passport.ts [hunk 2]
```diff
@@ -1,55 +1,51 @@
...
 
-export function login(email: string, password: string):Promise<any> {
+export function login(email: string, password: string): Promise<any> {
   return new Promise((resolve, reject) => {
     User.findOne({where: {email, password}})
-    .then(function(user) {
-      if (user === null) {
-        return reject(`User with provided email and password not exists`)
-      }
+    .then((user) => {
```
Comment (michalmikolajczyk) [inline src/helpers/passport.ts]: indent

File: src/helpers/passport.ts [hunk 3]
```diff
@@ -1,55 +1,51 @@
...
+      if (!user) return reject(`User with provided email and password not exists`);
```
Comment (michalmikolajczyk) [inline src/helpers/passport.ts]: cool one-liner, `/`/`` unneeded

File: src/services/auth/login.ts [hunk 4]
```diff
@@ -1,36 +1,36 @@
...
-        name: user.name,
-        email: user.email,
-      }
-    })
-  })
-  .catch(err => {
-    res.status(400).json({
-      err: true,
-      msg: `Invalid email or password`,
+    .then(({ token, user }) => {
```
Comment (michalmikolajczyk) [inline src/services/auth/login.ts]: great!

File: src/services/auth/login.ts [hunk 5]
```diff
@@ -1,36 +1,36 @@
...
+        err: false,
+        msg: `User logged in successfully`,
+        user: {
+          name: user.name,
+          email: user.email,
+        },
+      });
     })
-  })
+    .catch(err => {
```
Comment (michalmikolajczyk) [inline src/services/auth/login.ts]:
```
catch(err => res.status(400)
  .json({
    ...
  }))
```

sorry i hate semicolons 👍

File: src/services/auth/logout.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
-export default function logout(res, req, next) {
+export default function logout(res: any, req: any, next: any): void {
```
Comment (michalmikolajczyk) [inline src/services/auth/logout.ts]: is there a type for the requests?

File: src/services/auth/resend.ts [hunk 7]
```diff
@@ -1,36 +1,36 @@
...
+  User.findOne({ where: { email } })
+    .then((user) => {
+      signinEmail(email, user.dataValues.token)
+        .then(() => {
+            res.status(200).json({
+            err: false,
+            msg: 'Resend email successfully',
+          });
+        })
+        .catch(err => {
```
Comment (michalmikolajczyk) [inline src/services/auth/resend.ts]: perhaps the following block is not required, as there is a final `catch` for the chain?

File: src/services/auth/resend.ts [hunk 8]
```diff
@@ -1,36 +1,36 @@
...
+          res.status(500).json({
+            err: true,
+            msg: 'There was an error processing your request',
+          });
+        });
     })
     .catch(err => {
```
Comment (michalmikolajczyk) [inline src/services/auth/resend.ts]: I meant that one

File: src/services/auth/reset.ts [hunk 9]
```diff
@@ -1,37 +1,36 @@
-import { User } from '../../sequelize'
-import { resetEmail } from '../../helpers/nodemailer'
+import { User } from '../../sequelize';
+import { resetEmail } from '../../helpers/nodemailer';
 
-export default function reset(req, res, next) {
+export default function reset(req: any, res: any, next: any): void {
```
Comment (michalmikolajczyk) [inline src/services/auth/reset.ts]: why so `void`?

File: src/services/auth/reset.ts [hunk 10]
```diff
@@ -1,37 +1,36 @@
...
+  const { email } = req.body;
 
-  let { email } = req.body
+  User.findOne({ where: { email } })
```
Comment (michalmikolajczyk) [inline src/services/auth/reset.ts]: escape the void with a return

Comment (michalmikolajczyk) [reply src/services/auth/reset.ts]: return a promise

Comment (michalmikolajczyk) [reply src/services/auth/reset.ts]: you don't even have to keep it

File: src/services/auth/reset.ts [hunk 11]
```diff
@@ -1,37 +1,36 @@
...
+        })
+        .catch((err) => {
+          res.status(500).json({
+            err: true,
+            msg: 'There was an error with sending the reset link email',
+          });
+        });
     })
-    .catch(err => {
+    .catch((err) => {
```
Comment (michalmikolajczyk) [inline src/services/auth/reset.ts:38]: inner parentheses not required?

Comment (michalmikolajczyk) [reply src/services/auth/reset.ts:38]: like
```
.catch(err => res.status(...
```

File: src/services/auth/verify.ts [hunk 12]
```diff
@@ -1,55 +1,54 @@
...
+import { login } from '../../helpers/passport';
+import { User } from '../../sequelize';
+import { v1 } from 'uuid';
 
-export default function verify(req, res, next) {
+export default function verify(req: any, res: any, next: any): void {
+  const { token } = req.body;
 
-  let { token } = req.body
+  User.findOne({ where: { token } })
```
Comment (michalmikolajczyk) [inline src/services/auth/verify.ts]: as per previous comment

File: src/tests/auth/login.spec.ts [hunk 13]
```diff
@@ -1,69 +1,66 @@
...
+const should = chai.should();
+chai.use(chaiHttp);
 
-describe('user/login tests', function() {
+describe('user/login tests', () => {
 
-  before(function() {
-    return cleandb()
-  })
+  before(() => {
```
Comment (michalmikolajczyk) [inline src/tests/auth/login.spec.ts]:
```
before(cleandb))
```
?

Comment (michalmikolajczyk) [reply src/tests/auth/login.spec.ts]: or
```
before(() => cleandb())
```

File: src/helpers/passport.ts (hunk 1 above)
Comment (michalmikolajczyk) [inline src/helpers/passport.ts]: sooo this seems to be the only critical point? Perhaps also other promises.

Comment (jasiek-net) [inline src/helpers/passport.ts:39]: Here `passport` uses non-promise API, so I have to create and return my own `Promise` as it's described in an above article `Rookie mistake #4: using "deferred"`. In comment below I should omit my own `Promise` and return chain from `sequelize` request.

File: src/services/auth/login.ts (hunk 5 above)
Comment (jasiek-net) [reply src/services/auth/login.ts]: Ok, I will take care of this in future, but sometimes I have to log errors:
```
catch(err => {
  log(err)
  res.status(500).json({
    ...
  })
})
```

File: src/services/auth/logout.ts (hunk 6 above)
Comment (jasiek-net) [reply src/services/auth/logout.ts]: Yes, it's `{ Request, Response, NextFunction }` I'll fix this now.

File: src/services/auth/resend.ts (hunk 7 above)
Comment (jasiek-net) [reply src/services/auth/resend.ts]: Yes, but it's a two different errors (one from email client, one from wrong email address, so it's important to distinguish them)

File: src/services/auth/reset.ts (hunk 9 above)
Comment (jasiek-net) [reply src/services/auth/reset.ts]: We don't return anything, but `ts-lint` requires the type to be specified.

File: src/services/auth/reset.ts (hunk 10 above)
Comment (jasiek-net) [reply src/services/auth/reset.ts]: I want to write in a `servieces` functions that are `express middlewares` to have this code max reusable. All functions have parameters `req, res, next` and will return `void`.

File: src/services/auth/reset.ts (hunk 11 above)
Comment (jasiek-net) [reply src/services/auth/reset.ts:38]: It's not required only if we omit the parentheses https://github.com/airbnb/javascript#arrows--one-arg-parens

File: src/services/auth/login.ts (hunk 5 above)
Comment (jasiek-net) [reply src/services/auth/login.ts]: Already fixed

**Review by michalmikolajczyk — APPROVED** [review]

**Pull request merged**